.github/workflows/ci.yml

```diff
@@ -51,7 +51,7 @@ jobs:
           key: ${{ runner.os }}-cargo-build-target-${{ hashFiles('**/Cargo.lock') }}
 
       - name: Build
-        run: cargo build --all --verbose
+        run: cargo build --all --verbose --features strict
```

Comment:
The feature is a neat idea but why not just `-D warnings`?

Reply:
Good question... I originally implemented this by setting `RUSTFLAGS=-D warnings` and it looked like it built all of the dependencies with `-D warnings` too. (Then I copied the [feature approach from here](https://users.rust-lang.org/t/how-to-fail-a-build-with-warnings/2687/7)).

I didn't try passing -D directly to cargo though, I will try this and get back to you.

Reply:
Hey, sorry for the long turnaround on following this up!

* Can set `RUSTFLAGS` but this [builds all crate dependencies with deny warnings too](https://github.com/rust-lang/cargo/issues/10014), which seems excessive.
* Can't pass `-D warnings` directly to `cargo build`, only to `cargo rustc` - but I think this has the same shortcoming as setting `RUSTFLAGS`.

The best summary I've found (with links to some other resources) is the thread linked in the previous comment https://users.rust-lang.org/t/how-to-fail-a-build-with-warnings/2687